test/EFCacheTests/E2ETest.cs

```diff
@@ -48,6 +65,23 @@ static MyContext()
 
         protected override void OnModelCreating(DbModelBuilder modelBuilder)
         {
+            IDatabaseInitializer<MyContext> strategy;
+
+#if NET462
+            if (this.Database.Connection is SQLiteConnection)
+            {
+                strategy = new SqliteDropCreateDatabaseAlways<MyContext>(modelBuilder);
+            }
+            else
+            {
+#endif
+                strategy = new DropCreateDatabaseAlways<MyContext>();
+#if NET462
+            }
+#endif
```

Comment:
Please extract a private method - this way you will be able to clean up the `ifdefs` and simplify passing the strategy (i.e. you no longer need a variable)